### PR TITLE
[ENG-3905] rename open_badges to related_resource_types

### PR DESCRIPTION
**Pull request by aaxelb**
rename `open_badges` to `related_resource_types`, aligning with:
https://github.com/CenterForOpenScience/SHARE/pull/793
https://github.com/CenterForOpenScience/osf.io/pull/10004

-   Ticket: [ENG-3905]
-   Feature flag: n/a

## QA Notes

[ENG-3905]: https://openscience.atlassian.net/browse/ENG-3905?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by brianjgeiger** [comment]
Although the additional test is nice to have regardless.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
This just needs updating to use the `extra` object.

File: tests/engines/registries/unit/services/share-search-test.ts [hunk 1]
```diff
@@ -30,6 +30,13 @@ function getSearchResponse(identifiers?: string[]) {
                     sources: ['OSF', 'OSF Registries'],
                     subjects: [],
                     subject_synonyms: [],
+                    osf_related_resource_types: {
+                        data: true,
+                        papers: true,
+                        analytic_code: false,
+                        materials: false,
+                        supplements: false,
+                    },
```
Comment (brianjgeiger) [inline tests/engines/registries/unit/services/share-search-test.ts:39]: Does this need to be in the `extra` object as well?

Comment (jwalz) [reply tests/engines/registries/unit/services/share-search-test.ts:39]: SHARE pull it into the top-level dict in the elastic results: https://github.com/CenterForOpenScience/SHARE/pull/793/files#diff-3fd6a02be08431191d9e59f90d0f374c0541c2c13b8b16809d34809211e1eb63R125

**Review by brianjgeiger — APPROVED** [review]

**Pull request merged**